File: watch-app/SiWatchApp/Models/EventRecord.cs
using Newtonsoft.Json;
using SiWatchApp.Events;

namespace SiWatchApp.Models
{
    public class EventRecord : Record
    {
        public EventType EventType { get; set; }

        public LocationInfo Location { get; set; }

        public EventPriority Priority { get; set; }
        
        public override string ToString()
        {
            return $"EventRecord{{EventType={EventType},Timestamp={Timestamp},Value={Value},Location={Location}}}";
        }
    }
}

File: backend/facade/siwatch-device-api/src/main/java/ru/nord/siwatch/backend/facade/device/models/events/EventRecord.java
package ru.nord.siwatch.backend.facade.device.models.events;

import lombok.Getter;
import lombok.Setter;
import ru.nord.siwatch.backend.facade.device.models.Location;

import java.time.LocalDateTime;

@Getter @Setter
public class EventRecord
{
    private DeviceEventType eventType;
    private EventPriority priority;
    private Location location;
    private LocalDateTime timestamp;
    private Object value;
}
